core/src/main/kotlin/rys/ajaxpetproject/service/impl/ChatServiceImpl.kt

```diff
@@ -1,36 +1,122 @@
+        )
+            .then(chatRepository.addUser(userId, chatId))
+    }
 
-    override fun findAllChats(): List<MongoChat> = chatRepository.findAllBy()
+    override fun removeUser(userId: String, chatId: String): Mono<Unit> {
+        return Mono.`when`(
+            userService.findUserById(userId)
+                .switchIfEmpty{Mono.error(UserNotFoundException("User with id $userId not found"))},
+            findChatById(chatId)
+                .switchIfEmpty{Mono.error(ChatNotFoundException("Chat with id $chatId not found"))}
+        )
+            .then(chatRepository.removeUser(userId, chatId))
+    }
+
+    override fun addMessage(messageId: String, chatId: String): Mono<Unit> {
+        return Mono.`when`(
+            findChatById(chatId)
+                .switchIfEmpty{Mono.error(ChatNotFoundException("Chat with id $chatId not found"))},
+            messageService.findMessageById(messageId)
+                .switchIfEmpty{Mono.error(ChatNotFoundException("Message with id $messageId not found"))}
+        )
+            .then(chatRepository.addMessage(messageId, chatId))
+    }
 
-    override fun updateChat(id: ObjectId, updatedMongoChat: MongoChat): MongoChat {
+    override fun removeMessage(messageId: String, chatId: String): Mono<Unit> {
+        return chatRepository.removeMessage(messageId, chatId)
+    }
+
+    override fun delete(id: String): Mono<Unit> {
         return findChatById(id)
-            .let {
-                chatRepository.save(updatedMongoChat.copy(id = id))
-            }
+            .switchIfEmpty{Mono.error(ChatNotFoundException("Chat with id $id not found"))}
+            .then(chatRepository.delete(id))
     }
 
-    override fun deleteChat(id: ObjectId): Boolean {
-        findChatById(id)?.let {
-            chatRepository.deleteById(id)
-            return true
-        } ?: throw ChatNotFoundException("Chat not found")
+    override fun findAll(): Flux<MongoChat> {
+        return chatRepository.findAll()
     }
 
-    override fun deleteChats() = chatRepository.deleteAll()
+    override fun findChatsByUserId(userId: String): Flux<MongoChat> {
+        return chatRepository.findChatsByUserId(userId)
+    }
+
+    override fun getMessagesFromChatByUser(userId: String, chatId: String): Flux<MongoMessage> {
+        return Mono.`when`(
+            userService.findUserById(userId)
+                .switchIfEmpty{Mono.error(UserNotFoundException("User with id $userId not found"))},
+            findChatById(chatId)
+                .switchIfEmpty{Mono.error(ChatNotFoundException("Chat with id $chatId not found"))}
+        )
+            .thenMany(
+                getMessagesInChat(chatId).filter { it.userId == userId }
+            )
+    }
+
+    override fun getMessagesInChat(chatId: String): Flux<MongoMessage> {
+        return findChatById(chatId)
+            .switchIfEmpty{Mono.error(ChatNotFoundException("Chat with id $chatId not found"))}
```

Comment:
please, apply autoformatting to the whole file

Reply:
Got it, my bad